File: Semestrul 4/MPP/Proiecte/c#_gui/repository/Repository.cs
using System.Collections.Generic;

namespace csharp_avalonia.repository;
public interface Repository<T, ID>
{
    T save(T entity);
    
   // T findById(int id);
    T findOne(ID id);
    IEnumerable<T> findAll();
    
 //   void deleteById(int id);
    
  //  T existsById(int id);
    
}

File: Semestrul 4/MPP/Proiecte/REST-tema6/Persistance/src/main/java/persistance/Repository.java
package persistance;

import model.Entity;

import java.util.List;
import java.util.Optional;

public interface Repository<T, ID> {

    T save(T entity);

    T findOne(ID id);

    Iterable<T> findAll();

    T update(T entity);

    Optional<T> findById(ID id);

    T deleteById(ID id);

    Optional<T> existsById(ID id);
}
